Unity-Technologies/com.unity.multiplayer.samples.coop
# fix: leaving lobby when connection denied

### Description (*)

With this PR, the host removes clients from the lobby when their connection gets denied. Clients also leave on their side.
### Related Pull Requests

### Issue Number(s) (*)

Fixes issue(s): [MTT-2796](https://jira.unity3d.com/browse/MTT-2796)
### Manual testing scenarios

1. Start a host
2. Join with clients normally
3. Try to join with a client that gets their approval denied (currently not possible under normal circumstances, but can happen if you manually change CharSelectData.k_MaxLobbyPlayers to a value lower than the maximum number of players in the actual Lobby)
4. See that when you try to join again, you are allowed to and do not receive the "Player is already a part of the lobby" message
### Questions or comments

Something to note with this and with #515 is that if the host tries to remove a player at the same time as they are leaving, or vice-versa, it will generate an exception inside the Lobby service saying that it could not find the player. We may want to either catch and ignore that specific exception (not sure if it's a good idea) or maybe see with Lobby if a RemovePlayerAsync alternative could be added that doesn't generate exceptions in that case?
### Contribution checklist
